internal/ingress/controller/endpointslices.go

```diff
@@ -111,12 +112,38 @@ func getEndpointsFromSlices(s *corev1.Service, port *corev1.ServicePort, proto c
 				ports = append(ports, targetPort)
 			}
 		}
+		useTopologyHints = false
+		if zoneForHints != emptyZone {
+			useTopologyHints = true
+			// check if all endpointslices has zone hints
+			for _, ep := range eps.Endpoints {
+				if len(ep.Hints.ForZones) == 0 {
+					useTopologyHints = false
+					break
+				}
+			}
+		}
+		if useTopologyHints {
+			klog.V(3).Infof("All endpoint slices has zone hint, using zone %q for Service %q", zoneForHints, svcKey)
```

Comment:
you can move this log to after the for, inside the if, and get rid of this additional "if useTopologyHints" just for this logging

Reply:
All other possible places are inside loops, it will spam logs depending on count of endpointslices objects.